packages/grafana-runtime/src/services/live.ts

```diff
@@ -70,3 +72,18 @@ export const setGrafanaLiveSrv = (instance: GrafanaLiveSrv) => {
  * @public
  */
 export const getGrafanaLiveSrv = (): GrafanaLiveSrv => singletonInstance;
+
+/**
+ * Returns a promise that resolves with the {@link GrafanaLiveSrv} once it has
+ * connected
+ *
+ * @experimental
+ * @public
+ */
+export const getConnectedLiveSrv = (): Promise<GrafanaLiveSrv> => {
+  return new Promise(resolve => {
+    singletonInstance.addListener('connect', () => {
```

Comment:
what happens if it is already connected?

Reply:
should we remove the listener after it is resolved?

Reply:
Another good catch. Should add a check to see if we're already connected

Reply:
I like this design! Pushed a change